### DISCUSSION TITLE
su2+pyamg

### DISCUSSION THREAD

**Discussion by msahinae00** [su2code/SU2#2151] [Q&A]
Hi everyone

I have used both the early version of SU2 with pyAMG (python2) and the latest SU2-feature_adap branch with python3. We have even ported its mesh adaptation implementation to the GAMMA  mesh format due to geometry topology issues. 
 
However, there are two adapt options  in the python implementation (ADAP_BACK, ADAP_SOURCE) and l could not still figure out how to use it because  l don't know what kind of data file l should provide. For ADAP_BACK I have tried to use mesh files similar to the original volume mesh file with edges/corners in binary format (meshb) and it generally fails on complex geometries (with no ADAP_BACK option, it works nicely). Therefore, l am not sure whether l am providing the correct data file. This data file has to be only surface file with high order elements or just a volume mesh? 

ADAP_SOURCE is completely mistery to me. Not much explanation inside the python code. I should also say that l don't understand the following command either

    config_amg['options'] = config_amg['options'] + ' -inv-back'

Can someone please explain these plainly. I have also looked and feature_adapt Test files folder but unfortunately these options are closed and its data  files are missing.

**Comment by pcarruscag** [comment]
Moving to discussion since this is not part of the master/develop branch.
